main.py

```
@@ -53,219 +87,1126 @@
+    except Exception as e:
+        logger.error(f"Vertex AI analysis {analysis_id} failed: {e}")
+        error_response = AnalysisResponse(
+            analysis_id=analysis_id,
+            status="failed",
+            results={"error": str(e)},
+            timestamp=datetime.now(),
+            processing_time_seconds=(datetime.now() - start_time).total_seconds()
         )
-        
+        analysis_storage[analysis_id] = error_response.dict()
+        raise HTTPException(status_code=500, detail=f"Analysis failed: {str(e)}")
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)